lnd.go

```diff
@@ -1633,102 +1624,87 @@ func initializeDatabases(ctx context.Context,
 		}
-
-		closeFuncs = append(closeFuncs, func() {
-			dbs.graphDB.Close()
-		})
-
-		dbs.chanStateDB = dbs.graphDB
-	} else {
-		ltndLog.Infof("Database replication is available! Creating " +
-			"local and remote channeldb instances")
-
-		// Otherwise, we'll open two instances, one for the state we
-		// only need locally, and the other for things we want to
-		// ensure are replicated.
-		dbs.graphDB, err = channeldb.CreateWithBackend(
-			databaseBackends.GraphDB,
-			channeldb.OptionSetRejectCacheSize(cfg.Caches.RejectCacheSize),
-			channeldb.OptionSetChannelCacheSize(cfg.Caches.ChannelCacheSize),
-			channeldb.OptionSetBatchCommitInterval(cfg.DB.BatchCommitInterval),
-			channeldb.OptionDryRunMigration(cfg.DryRunMigration),
-		)
-		switch {
-		// As we want to allow both versions to get thru the dry run
-		// migration, we'll only exit the second time here once the
-		// remote instance has had a time to migrate as well.
-		case err == channeldb.ErrDryRunMigrationOK:
-			ltndLog.Infof("Local DB dry run migration successful")
-
-		case err != nil:
-			err := fmt.Errorf("unable to open local channeldb: %v", err)
-			ltndLog.Error(err)
-			return nil, nil, err
+		cleanUp = func() {
+			for name, closeFunc := range closeFuncs {
+				if err := closeFunc(); err != nil {
+					ltndLog.Errorf("Error closing %s "+
+						"database: %v", name, err)
+				}
+			}
 		}
+	)
+	if databaseBackends.Remote {
+		ltndLog.Infof("Using remote %v database! Creating "+
+			"graph and channel state DB instances", cfg.DB.Backend)
+	} else {
+		ltndLog.Infof("Creating local graph and channel state DB " +
+			"instances")
+	}
+
+	// Otherwise, we'll open two instances, one for the state we only need
+	// locally, and the other for things we want to ensure are replicated.
+	dbs.graphDB, err = channeldb.CreateWithBackend(
+		databaseBackends.GraphDB,
+		channeldb.OptionSetRejectCacheSize(cfg.Caches.RejectCacheSize),
+		channeldb.OptionSetChannelCacheSize(cfg.Caches.ChannelCacheSize),
+		channeldb.OptionSetBatchCommitInterval(cfg.DB.BatchCommitInterval),
+		channeldb.OptionDryRunMigration(cfg.DryRunMigration),
+	)
+	switch {
+	// Give the DB a chance to dry run the migration. Since we know that
+	// both the channel state and graph DBs are still always behind the same
+	// backend, we know this would be applied to both of those DBs.
+	case err == channeldb.ErrDryRunMigrationOK:
+		ltndLog.Infof("Graph DB dry run migration successful")
+		return nil, nil, err
 
-		closeFuncs = append(closeFuncs, func() {
-			dbs.graphDB.Close()
-		})
-
-		ltndLog.Infof("Opening replicated database instance...")
-
-		dbs.chanStateDB, err = channeldb.CreateWithBackend(
-			databaseBackends.ChanStateDB,
-			channeldb.OptionDryRunMigration(cfg.DryRunMigration),
-			channeldb.OptionSetBatchCommitInterval(cfg.DB.BatchCommitInterval),
-		)
-		switch {
-		case err == channeldb.ErrDryRunMigrationOK:
-			return nil, nil, err
+	case err != nil:
+		cleanUp()
```

Comment:
If this is the only place where the non-decorated  closes are called, isn't it more clear to just write out the non-decorated closes here and skip the close func map?